src/event-detail/ui/event-detail.controller.js

```diff
@@ -39,24 +39,26 @@ function EventDetail(
     $scope.eventId = offerLocation;
 
     var offer = udbApi.getOffer(offerLocation);
-    var permission = udbApi.hasPermission(offerLocation);
+
+    var permissions = udbApi.getUserPermissions(offerLocation);
 
     offer.then(showOffer, failedToLoad);
 
-    $q.all([permission, offer])
+    $q.all([permissions, offer])
       .then(grantPermissions);
   }, function() {
     $location.path('404');
   });
 
   /**
-   * Grant permissions based on permission-data.
+   * Grant permissions based on permissions-data.
    * @param {Array} permissionsData
    *  The first array-item is assumed to be true, if the user is not owner the permission check rejects.
    *  The second value holds the offer itself.
    */
   function grantPermissions(permissionsData) {
-    var hasPermission = permissionsData[0];
+    var permissions = permissionsData[0];
+    var hasPermissions = permissionsData[0].length > 0;
```

Comment:
Since you already stored permissionsData[0] in a variable, you can call it like this :) 
```suggestion
    var hasPermissions = permissions.length > 0;
```